Sources/Prelude/Predicate.swift

```diff
@@ -0,0 +1,62 @@
+//
+// This source file is part of Prelude, an open source project by Wayfair
+//
+// Copyright (c) 2018 Wayfair, LLC.
+// Licensed under the 2-Clause BSD License
+//
+// See LICENSE.md for license information
+//
+
+public struct Predicate<A> {
+    public let contains: (A) -> Bool
+}
+
+// MARK: - set algebra
+
+public extension Predicate {
+    func intersection(_ other: Predicate) -> Predicate {
+        return .init { self.contains($0) && other.contains($0) }
+    }
+
+    func subtracting(_ other: Predicate) -> Predicate {
+        return .init { self.contains($0) && !other.contains($0)
+        }
+    }
+
+    func symmetricDifference(_ other: Predicate) -> Predicate {
+        return .init { element in
+            return self.contains(element) && !other.contains(element) ||
+                !self.contains(element) && other.contains(element)
+        }
+    }
+
+    func union(_ other: Predicate) -> Predicate {
+        return .init { self.contains($0) || other.contains($0) }
+    }
+}
+
+public extension Predicate {
+    var inverse: Predicate<A> {
+        return .init { !self.contains($0) }
+    }
+
+    init<S>(_ sequence: S) where S: Sequence, A: Equatable, A == S.Element {
```

Comment:
I’m not sure we want this initializer. It’s on `SetAlgebra` and it’s possible to implement kinda, but maybe it’s best if we don’t have it?

Reply:
Agreed, thanks! I’ll remove it.